### PR TITLE
Fixed unnecessary callbacks in dropdown

### PR DESCRIPTION
**Pull request by hemish11**
This PR includes a fix for the https://github.com/EnsembleUI/ensemble/issues/877 in which there were unnecessary onChanged callbacks happening when the same item was selected in the dropdown which was previously selected.

### CODE REVIEW

**Review by kmahmood74 — APPROVED** [review]
fix the formatting and unnecessary changes and then checkin. Already approved so you don't need another approval. Also make sure to close the associated ticket after you merge and close this PR

**Comment by hemish11** [comment]
@kmahmood74 Please review it now and also added issue number in first message

**Review by kmahmood74 — APPROVED** [review]
approved, please go ahead and merge and commit. Also remember to close this PR and the associated ticket once you are done

**Pull request merged**
